ProfilerGUI/Source/Shared/SystemUtils.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace ProfilerGUI.Source.Shared
+{
+    /// <summary>
+    /// System utility functions.
+    /// </summary>
+    public static class SystemUtils
+    {
+        /// <summary>
+        ///  Sets the given environment variable on the 'Process' level, unless it is already configured. I.e. this process and any of its children
+        ///  will see the environment variable. Returns <code>true</code>
+        ///  if changes were made, i.e. the variable was not already set to the target value.
+        /// </summary>
+        public static bool SetEnvironmentVariable(string variableName, string value)
+        {
+            if (!Equals(Environment.GetEnvironmentVariable(variableName, EnvironmentVariableTarget.Process), value))
+            {
+                Environment.SetEnvironmentVariable(variableName, value, EnvironmentVariableTarget.Process);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Asynchronously runs the given executable with the given command line arguments, working dir and environment.
+        /// Does not enable process events.
+        /// </summary>
+        public static Process RunNonBlocking(string processPath, string args, string workingDir, List<(string, string)> environmentVariables)
+        {
+            Process process = new Process
+            {
+                StartInfo = CreateStartInfo(processPath, args, workingDir, environmentVariables)
+            };
+            process.Start();
+            return process;
+        }
+
+        private static ProcessStartInfo CreateStartInfo(string processPath, string args, string workingDir, List<(string, string)> environmentVariables)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                FileName = processPath,
+                Arguments = args
+            };
+
+            startInfo.UseShellExecute = false;
+
+            foreach (var (environmentVariable, value) in environmentVariables)
```

Comment:
i'd explicitly write (string .., string..) according to guidelines

Reply:
thanks.I tried that but all my attempts were refused by the compiler so I gave up and used var ;-)